theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-toolbar-contribution.tsx

```diff
@@ -82,6 +83,18 @@ export class TraceViewerToolbarContribution implements TabBarToolbarContribution
                 signalManager().fireResetZoomSignal();
             }
         });
+        registry.registerCommand(
+            TraceViewerToolbarCommands.OPEN_NEW_TRACE, {
+            isVisible: (w: Widget) => {
+                if (w instanceof TraceExplorerOpenedTracesWidget) {
+                    return true;
+                }
+                return false;
+            },
+            execute: () => {
+                signalManager().fireOpenNewTraceSignal();
```

Comment:
you can just call the other open command: registry.executeCommand(OpenTraceCommand.id); No need for the signal.

Reply:
I understand now why the it was moved to the place holder, i.e. to execute the health check as well. So, the command handler of the open command should to the heath check as part of the command execution. That means the code from the placeholder widget has to move into the command-handler. Then the command can be executed from multiple places and the behavior is the same.

Reply:
@bhufmann Command handler of the open command now does the health check as well. Works well, but not sure if my reorganization is 100% correct